esp/esp/program/modules/handlers/schedulingcheckmodule.py

```diff
@@ -553,3 +554,33 @@ def special_classroom_types(self):
          return self.formatter.format_table(mismatches,
                                             {'headings': HEADINGS},
                                             help_text=self.special_classroom_types.__doc__)
+
+     # This isn't really a scheduling check. It's a check that's useful
+     # to run before scheduling. But it works well with the format and
+     # this way everyone else doesn't have to rediscover the round_to
+     # argument to ESPUser.getTaughtTime() every year.
+     def hosed_teachers(self):
+         """
+         Teachers who have registered almost as many hours of classes
+         as hours of availability. Intended to be run before scheduling,
+         and will not change as classes are scheduled.
+         """
+         teachers = self.p.teachers()['class_submitted']
+         hosed = []
+         for teacher in teachers:
+             # This will break if we ever start having class blocks
+             # that aren't an hour long
+             availability = len(teacher.getAvailableTimes(self.p, ignore_classes=True))
```

Comment:
I think making this generalize well would be more tricky than I feel like bothering with. None of the big chapters have blocks that aren't an hour long for their Splashes, and distinguishing cases like Stanford's 45-minute-long "hours" from hypothetical 30-minute or 90-minute blocks seems annoying when we really do want to be working in whole hours. Things will get incredibly messy if we try to work with real lengths because of the thing where long classes aren't n*50 minutes long, they're n*60-10.